lib/model-builder.js

```diff
@@ -355,6 +355,11 @@ ModelBuilder.prototype.define = function defineClass(className, properties, sett
       forceId = false;
     }
 
+    // Make id value mandatory if it is not autogenerated
+    if (!idProp.generated) {
+      ModelClass.validatesPresenceOf(idName);
+    }
```

Comment:
There are three possible values of `generated` field:

- `undefined` - the user did not indicate whether the PK values should be generated by the DB or provided by the caller, expecting the framework to pick a reasonable default
- `true` - the user wants PK values to be generated by the database
- `false` - the user wants PK values to be provided by the caller

As LB user, I would expect the default behavior to be `generated: true` for primary keys. I.e. when I don't provide any `generated` flag, the framework should assume `generated: true`.

Would that reduce the number of tests broken by your change(s)?

Reply:
Proposed implementation to try:

```js
if (idProp.generated === false) {
  ModelClass.validatesPresenceOf(idName);
}
```